### PR TITLE
feat: Correctly define all product module models and SQL migration script

### PR DESCRIPTION
**Pull request by sradevski**
The only thing remaining is modifying `upsertWithReplace` to handle many-to-many constraints correctly and not cascade updates to them. This is something that we should do separately though.

### CODE REVIEW

File: .changeset/brave-beds-join.md
Comment (olivermrbl) [inline .changeset/brave-beds-join.md]: todo: Doesn't have to be a minor - can be a patch

**Review by olivermrbl — APPROVED** [review]

File: packages/product/src/models/product-category.ts
Comment (adrien2p) [inline packages/product/src/models/product-category.ts:106]: todo: `deleted_at: Date | null`

File: packages/product/src/models/product-collection.ts
Comment (adrien2p) [inline packages/product/src/models/product-collection.ts:70]: suggestion: line 66 `deletes_at: Date | null`

File: packages/product/src/models/product-image.ts
Comment (adrien2p) [inline packages/product/src/models/product-image.ts:25]: suggestion: rm

File: packages/product/src/models/product-image.ts
Comment (adrien2p) [inline packages/product/src/models/product-image.ts:29]: comment: In general I prefer having the index on the property itself, unless it is a multi columns, but I don't have a strong opinion but since this is what we are doing everywhere I d like to know why we changed that in this pr :)

File: packages/product/src/models/product-collection.ts
Comment (adrien2p) [reply packages/product/src/models/product-collection.ts:70]: same for all models

**Review by adrien2p — COMMENTED** [review]
LGTM few comments a bit after the merge 😂 I suppose the rest will be handle later right?

File: packages/product/src/models/product-image.ts
Comment (sradevski) [reply packages/product/src/models/product-image.ts:29]: @adrien2p I was aiming for consistency on where we define the index, regardless if it is multi-column or not. Like that, it's easier to find what indices have been defined for a model (just by looking at the top). That is also in-line with how postgres presents the indices if you do `/d` for a table, for example.

File: packages/product/src/models/product-category.ts
Comment (sradevski) [reply packages/product/src/models/product-category.ts:106]: Ah true, I'll take a note on this

**Pull request merged**
